File: frontend/ui/src/types/Product.ts
import { Supplier } from './Supplier';

export interface Product {
  id: number;
  barcode: string;
  name: string;
  purchasePrice: number;
  salePrice: number;
  stock: number; // Changed from stockQuantity
  minStock: number; // Changed from minStockAlert
  lowStock: boolean; // New field
  unitOfMeasure?: 'KG' | 'LITER' | 'UNIT'; // Made optional
  supplier: Supplier | null;
}

File: backend/pos/src/main/java/mx/bepos/pos/domain/Product.java
package mx.bepos.pos.domain;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "products")
public class Product {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(unique = true, length = 50)
    private String barcode;

    @Column(nullable = false, length = 150)
    private String name;

    @Column(precision = 10, scale = 2)
    private BigDecimal purchasePrice;

    @Column(nullable = false, precision = 10, scale = 2)
    private BigDecimal salePrice;

    @Column(precision = 10, scale = 3)
    private BigDecimal stockQuantity;

    @Column(precision = 10, scale = 3)
    private BigDecimal minStockAlert;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 10)
    @Builder.Default
    private UnitOfMeasure unitOfMeasure = UnitOfMeasure.UNIT;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "supplier_id")
    private Supplier supplier;
}
